src/renderers/dom/server/ReactMarkupChecksum.js

```diff
@@ -25,11 +27,15 @@ var ReactMarkupChecksum = {
   addChecksumToMarkup: function(markup) {
     var checksum = adler32(markup);
 
-    // Add checksum (handle both parent tags and self-closing tags)
-    return markup.replace(
-      TAG_END,
-      ' ' + ReactMarkupChecksum.CHECKSUM_ATTR_NAME + '="' + checksum + '"$&'
-    );
+    // Add checksum (handle both parent tags, comments and self-closing tags)
+    if (markup.match(COMMENT_START)) {
```

Comment:
Let's make this `.test` - we don't need the match results. Otherwise this looks good, thanks!